src/sas/qtgui/MainWindow/DataExplorer.py

```diff
@@ -1883,15 +1893,17 @@ def closePlotsForItem(self, item):
 
         pass  # debugger anchor
 
-    def onAnalysisUpdate(self, new_perspective=""):
+    def onAnalysisUpdate(self, new_perspective_name: str):
```

Comment:
Do we no longer need the default parameter?

Reply:
"" isn't the name of any perspective, and a valid name must be provided. Better to fix the signature to (str) -> None and force those using it to supply a name.

As an aside, I tried to make this system not depend on strings, but on class names, but it was such a major change I stashed it.